ui/src/literature/containers/CitationSummaryBoxContainer.jsx

```diff
@@ -11,10 +11,8 @@ const stateToProps = state => ({
 
 const dispatchToProps = dispatch => ({
   onQueryChange(query) {
-    dispatch(fetchCitationSummary(query));
+    dispatch(fetchCitationSummary(query.toJS()));
   },
 });
 
-export default connect(stateToProps, dispatchToProps)(
-  convertAllImmutablePropsToJS(CitationSummaryBox)
-);
+export default connect(stateToProps, dispatchToProps)(CitationSummaryBox);
```

Comment:
can we just use ``convertAllImmutablePropsToJS``?

Reply:
no this is the whole point, we need to pass immutable map which is used for equality check, otherwise these mysterious rerenders ends up generating different object references of the same query and ends up triggering `onQueryChange` callback.